### PR TITLE
Pack WarnAsError should fail the pack creation and delete the existing package from disk

### LINKED ISSUE #7404: Honor TreatWarningsAsErrors during NuGet Package creation
From an internal thread;

```

E:\A\_work\3324\s\bin\2\win-x86\dotnet\sdk\2.1.500-preview-009365\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(202,5): error NU5118: File 'E:\A\_work\3324\s\TestAssets\TestPackages\dotnet-portable-v1-prefercli\bin\Release\netcoreapp1.1\dotnet-portable-v1-prefercli.runtimeconfig.json' is not added because the package already contains file 'lib\netcoreapp1.1\dotnet-portable-v1-prefercli.runtimeconfig.json' [E:\A\_work\3324\s\TestAssets\TestPackages\dotnet-portable-v1-prefercli\dotnet-portable-v1-prefercli.csproj] [E:\A\_work\3324\s\build.proj]
2018-10-13T10:52:25.2756630Z            Successfully created package 'E:\A\_work\3324\s\/bin/2/win-x86/test//packages/dotnet-portable-v1-prefercli.1.0.0.nupkg'.

```

The package is still created even though it's elevated to an error. 

Note that the exit code should not be 0!

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Home/issues/7404
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: 

This is a common issue for both pack entry points. 
We silently ignore the warnings elevated to errors. We don't even account for them in the pack command result. 

This change adds an API that can account for failure caused elevating a warning to an error.

## Testing/Validation

Tests Added: Yes
Reason for not adding tests: 
Validation:  Manual + automation

### CODE REVIEW

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 using System;
 using System.Diagnostics;
+using System.Linq;
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs]: nit: it doesn't appear that this using is needed.

File: src/NuGet.Core/NuGet.Packaging/Core/PackagingException.cs [hunk 2]
```diff
@@ -39,4 +39,4 @@ public virtual ILogMessage AsLogMessage()
             return _logMessage;
         }
     }
-}
\ No newline at end of file
+}
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.Packaging/Core/PackagingException.cs]: as much as I love files ending with a new line, you haven't made any other modifications to this file.

**Review by zivkan — APPROVED** [review]

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs]: yeah good catch.
The namespace was from the older implementation. Fixed.

File: src/NuGet.Core/NuGet.Packaging/Core/PackagingException.cs (hunk 2 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Packaging/Core/PackagingException.cs]: reverted.

**Pull request merged**